android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java

```diff
@@ -1180,27 +1179,24 @@ private void applyTouchFeedback(Integer rippleColor)
 
 		// Fetch the background drawable that we'll be applying the ripple effect to.
 		Drawable backgroundDrawable = this.background;
-		if (backgroundDrawable == null) {
-			backgroundDrawable = this.nativeView.getBackground();
+		if (backgroundDrawable instanceof TiBackgroundDrawable) {
+			final TiBackgroundDrawable drawable = (TiBackgroundDrawable) backgroundDrawable;
+
+			backgroundDrawable = drawable.getBackground();
 		}
+		if (backgroundDrawable instanceof PaintDrawable) {
+			final PaintDrawable drawable = (PaintDrawable) backgroundDrawable;
+
+			if (drawable.getPaint().getColor() == Color.TRANSPARENT) {
```

Comment:
It would be better to check for transparency like the below in case the RGB components are non-zero.
```java
if (Color.alpha(drawable.getPaint().getColor()) <= 0) {
}
```